atlp-rwanda/atlp-pulse-mobile
# Ft(#84): Fetch Trainee Notifications

### What does this PR do?
This PR implements the functionality to fetch trainee notifications from the database using GraphQL queries and mutations. It enables real-time updates for notifications and allows trainees to mark notifications as read, improving user experience and data efficiency.

### Description of Task to be completed

- Fetch Trainee Notifications: Created a GET_TRAINEE_NOTIFICATIONS GraphQL query to retrieve notifications specific to trainees. This query includes fields such as id, message, createdAt, read status, and sender information.
 
- Mark Notifications as Read: Implemented a MARK_NOTIFICATION_AS_READ mutation to allow trainees to mark notifications as read. This mutation updates the notification’s read status in the database.

- Real-Time Notifications: Added a TRAINEE_NOTIFICATION_SUBSCRIPTION subscription for real-time notification updates. When a new notification is sent to a trainee, it is immediately pushed to their UI.

- Frontend Integration: Integrated the GraphQL query, mutation, and subscription with the frontend to display notifications dynamically, handle real-time updates, and manage unread notification counts. Enhanced UI components to show notifications efficiently and updated styling where needed.

### How should this be manually tested?

- Setup: Ensure the backend server with GraphQL support is running and accessible.

### Fetch Notifications:

- Log in as a trainee and navigate to the notifications page.
- Verify that notifications are fetched and displayed correctly from the database.

### Mark as Read:

- Click on an unread notification to mark it as read.
- Confirm that the notification’s read status updates in the database, and the unread count in the UI decreases accordingly.

### Error Handling: 
Test with no internet connection and observe that appropriate error messages are displayed.

### Screenshots: 

<div style="display: flex; flex-wrap: wrap;">
    <img src="https://github.com/user-attachments/assets/8d39e83d-bd69-4e96-a157-bb788f80f73f" alt="Screenshot 1" height="480px">
    <img src="https://github.com/user-attachments/assets/3e808931-d357-444a-b46b-f89f69948f28" alt="Screenshot 1" height="480px">
    <img src="https://github.com/user-attachments/assets/d383cf4a-83f1-4617-bfe6-fa1197f7454f" alt="Screenshot 2" height="480px">
</div>
